src/main/database.cpp

```diff
@@ -139,6 +140,12 @@ void Database::openLockFile() {
 }
 
 void Database::initAndLockDBDir() {
+    if (DBConfig::isDBPathInMemory(databasePath)) {
+        if (dbConfig.readOnly) {
+            throw Exception("Cannot open an in-memory database under READ ONLY mode.");
```

Comment:
cover or escape. See if u want to escape across other places in this PR

Reply:
it's covered in tests. but codecov is not running under in-mem mode, I guess that's why codecov is not reporting them as covered here.